### ISSUE TITLE
CredentialsApi types does not satisfy the constraints

### ISSUE THREAD

**Issue by oadamovskyi** [hyperledger/aries-framework-javascript#1205]
Hey

I tried to create module of agent initialization class for further usage it in my Node server, but when I tries compile from **.ts** to **.js** with `tsc --esModuleInterop true <file path>` I got following error:
```
node_modules/@aries-framework/core/build/modules/credentials/CredentialsApiOptions.d.ts:94:49 - error TS2344: Type 'CFsFromCPs<CPs>' does not satisfy the constraint 'CredentialFormat[]'.
  Type '[] | _CFsFromCPs<CPs>' is not assignable to type 'CredentialFormat[]'.
    Type '_CFsFromCPs<CPs>' is not assignable to type 'CredentialFormat[]'.
      Type 'FlatArray<{ [CP in keyof CPs]: ExtractCredentialFormats<ExtractCredentialFormatServices<CPs[CP]>>; }>' is not assignable to type 'CredentialFormat'.
        Type 'unknown' is not assignable to type 'CredentialFormat'.

94     credentialFormats?: CredentialFormatPayload<CFsFromCPs<CPs>, 'acceptRequest'>;
```

It is interesting because using `ts-node <file path> `everything works ok and as I understand` ts-node` uses `tsc` underneath

Tried to compile the next code: 
```
import type { InitConfig } from '@aries-framework/core'

import { Agent, AutoAcceptCredential, AutoAcceptProof, HttpOutboundTransport } from '@aries-framework/core'
import { agentDependencies, HttpInboundTransport } from '@aries-framework/node'

const bcovrin = //hardcoded genesis to Bcovrin Test Network

export class BaseAgent {
  public port: number
  public name: string
  public config: InitConfig
  public agent: Agent

  public constructor(port: number, name: string) {
    this.name = name
    this.port = port

    const config: InitConfig = {
      label: name,
      walletConfig: {
        id: name,
        key: name,
      },
      publicDidSeed: 'test0000000000000000000000000000',
      indyLedgers: [
        {
          genesisTransactions: bcovrin,
          id: 'testt' + name,
          indyNamespace: 'testt' + name,
          isProduction: false,
        },
      ],
      endpoints: [`http://localhost:${this.port}`],
      autoAcceptConnections: true,
      autoAcceptCredentials: AutoAcceptCredential.ContentApproved,
      autoAcceptProofs: AutoAcceptProof.ContentApproved,
    }

    this.config = config

    this.agent = new Agent({ config, dependencies: agentDependencies })
    this.agent.registerInboundTransport(new HttpInboundTransport({ port }))
    this.agent.registerOutboundTransport(new HttpOutboundTransport())
  }

  public async initializeAgent() {
    await this.agent.initialize()
  }
}
```

**Comment by TimoGlastra** [comment]
Thanks for opening this issue @oadamovskyi! I can't immediately see what the problem could be.

Could you share your package.json, yarn.lock (or package-lock.json) and tsconfig file? Or, even beter, share a runnable example in a git repo?

**Comment by oadamovskyi** [comment]
@TimoGlastra here is the [repo](https://github.com/oadamovskyi/afj_compiling_issue_sample). I`ve been using `tsc BaseAgent.ts` to test
Thanks :-)

**Comment by simonwh** [comment]
Having the same issue :)

**Comment by TimoGlastra** [comment]
I've been able to reproduce the issue (and discovered a few more issues with the typescript output). 

I'll add a check to the AFJ CI to make sure we catch these types of errors. For now if you enable `skipLibCheck` in your tsconfig.json you should be good (which isn't ideal of course), and I'll try to resolve this issue as soon as possible. Thank you for bringing this to our attention!

**Comment by TimoGlastra** [comment]
After doing some more digging it seems this issue is present in TypeScript 4.9.4 (which I've used for testing this locally), but it's not present in the TS version used by the AFJ repo (4.3.x). I've added a check to the repo, but it passes.

I've also tried 4.4.x, 4.5.x, 4.6.x, 4.7.x, 4.8.x which all seem to work fine, but it breaks on 4.9.x. So for now you could also set your ts version at 4.8.x and it should work fine.

You will have to install @types/express. I've fixed this and will make a PR so you won't have to install it.

**Comment by TimoGlastra** [comment]
@oadamovskyi I've made a PR that should fix the issue in #1214. Let's hope this will resolve your issue

**Comment by oadamovskyi** [comment]
@TimoGlastra You helped me so much, thanks 
P.S *emoji with a little tear of happiness and sincere smile*
